src/Raven.Server/Documents/Indexes/Static/DynamicArray.cs

```diff
@@ -840,5 +840,62 @@ IEnumerator IEnumerable.GetEnumerator()
                 return GetEnumerator();
             }
         }
+
+        private class LazyStringEqualityComparer : IEqualityComparer<object>
```

Comment:
I think the name might be a bit misleading. It's not the comparer which is supposed to be used to compare LazyStringValues only. It's a comparer of any objects which has special behavior of dealing with string and LSV. Also currently it's used for distinct purposes only. Maybe `LazyStringAwareDistinctComparer`?